chain-connect/src/chainApis/GalaChainFeeApi.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  DryRunParams,
+  DryRunResponseBody,
+  FeeAuthorizationParams,
+  FeeAuthorizationResponseBody,
+  FeeProperties,
+  FetchFeeAuthorizationsParams,
+  FetchFeeAuthorizationsResponseBody,
+  FetchFeePropertiesParams,
+  GalaChainResponse,
+  SetFeePropertiesParams
+} from "@gala-chain/api";
+
+import { GalaChainProvider } from "../GalaChainClient";
+
+export class GalaChainFeeApi {
+  constructor(
+    private chainCodeUrl: string,
+    private connection: GalaChainProvider
+  ) {}
+
+  public AuthorizeFee(dto: FeeAuthorizationParams) {
+    return this.connection.submit<GalaChainResponse<FeeAuthorizationResponseBody>, FeeAuthorizationParams>({
+      method: "AuthorizeFee",
+      payload: dto,
+      sign: true,
+      url: this.chainCodeUrl
+    });
+  }
```

Comment:
Is it possible to rework these methods to use DTOs rather than adding a `*Params` type for each? 

I would think that untyped data could be converted into a DTO class instance using `createValidDTO()`, and serialized using either the  `dto.serialize()` instance method or using `instanceToPlain` from the `class-validator` npm module to convert it into an object suitable for submitting to chain - although I haven't worked in the `connect` module much yet, I imagine I could be missing something.

Reply:
This is a more detailed illustration of what I meant in my previous comment:

https://github.com/GalaChain/sdk/pull/400

Basically, working with the types that already exist, in a style that works similarly to how the `chain-api` module was built. 

The `@gala-chain/connect` library already depends on `@gala-chain/api`, so might as well also use `BigNumber`, `class-validator`, and `class-transformer`.